.github/workflows/ok-to-test.yml

```diff
@@ -0,0 +1,55 @@
+# File based on https://github.com/imjohnbo/ok-to-test/blob/master/.github/workflows/ok-to-test.yml
+name: Ok To Test
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  ok-to-test:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      issues: write
+      contents: write
+
+    # Only run for PRs, not issue comments
+    if: ${{ github.event.issue.pull_request }}
+    steps:
+      - name: Resolve PR
+        id: resolve-pr
+        uses: actions/github-script@v3
+        with:
+          script: |
+              const request = {
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  pull_number: context.issue.number
+              }
+              core.info(`Getting PR #${request.pull_number} from ${request.owner}/${request.repo}`)
+              try {
+                  const result = await github.pulls.get(request)
+                  return result.data
+              } catch (err) {
+                  core.setFailed(`Request failed with error ${err}`)
+              }
+      - name: Print pr data
+        run: |
+            echo head repository is ${{ fromJSON(steps.resolve-pr.outputs.result).head.repo.full_name }}
+            echo head branch is ${{ fromJSON(steps.resolve-pr.outputs.result).head.ref }}
+            echo head sha is ${{ fromJSON(steps.resolve-pr.outputs.result).head.sha }}
+
+      - name: slash command dispatch
+        uses: peter-evans/slash-command-dispatch@a28ee6cd74d5200f99e247ebc7b365c03ae0ef3c # v3.0.1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          reaction-token: ${{ secrets.GITHUB_TOKEN }}
+          issue-type: pull-request
+          commands: ok-to-test
+          static-args: |
+            PR_NUMBER=${{ github.event.issue.number }}
+            HEAD_REPO=${{ fromJSON(steps.resolve-pr.outputs.result).head.repo.full_name }}
+            HEAD_BRANCH=${{ fromJSON(steps.resolve-pr.outputs.result).head.ref }}
```

Comment:
setting the ref that is checked out this way opens us up to security vulnerabilities. We can use the event payload to avoid that `github.event.client_payload.slash_command.args.named.sha`.

Without this an approver can mark a pr as `/ok-to-test` then an attacker can push a malicious commit to the PR that will be run. The purpose of the `/ok-to-test` is that an approver can approve an exact set of changes to ensure there's nothing malicious run with our secrets. Using the sha method makes that attack impossible. https://github.com/Azure/aks-app-routing-operator/blob/1028b62d138483de6dfca64c54ed39c11c54c50f/.github/workflows/pr-validate-fork.yaml#L42 Here's an example of how we do this in another repo. We have quite a few checks against the sha.

Reply:
@OliverMKing so you are suggesting to add the following if conditions in the workflow and so approver has to specify which sha to use for the workflows correct ? ex: ```/ok-to-test sha=123456```. But why do you say without specifying sha it would cause security vulnerabilities ?

```
if:
      github.event_name == 'repository_dispatch' &&
      github.event.client_payload.slash_command.args.named.sha != '' &&
      contains(github.event.client_payload.pull_request.head.sha, github.event.client_payload.slash_command.args.named.sha)
```

 @cxznmhdcxz made changes to get the latest sha for that run and comment /ok-to-test when approver has "write" access only for that one time run. Approver has to make sure the latest sha commit doesn't have malicious code. Please correct If my understanding is wrong.

Reply:
@OliverMKing Thanks for explaining, it would be a very short span of time though we can't compromise security for it. @cxznmhdcxz Ruijun this needs to be changed back accept ```/ok-to-test sha=123456``` with sha arguments. Please let us know if you have any questions or concerns.